scripts/globals/spells/sleepga.lua

```diff
@@ -17,8 +17,20 @@ function onSpellCast(caster,target,spell)
     local mINT = target:getStat(MOD_INT);
     local dINT = (pINT - mINT);
     local resm = applyResistanceEffect(caster,spell,target,dINT,ENFEEBLING_MAGIC_SKILL,0,EFFECT_SLEEP_I);
+
+    if (caster:isMob()) then
+        if (caster:getPool() == 5310) then -- Amnaf (Flayer)
+            caster:resetEnmity(target);
+        end
+        -- Todo: get rid of this whole block by handling it in the mob script
+        -- this requires a multi target enmity without specifying a target (have to get hate list from mob)
+        -- OR by altering onSpellPrecast to have a target param..
+        -- onMonsterMagicPrepare is not a realistic option.
+        -- You'd have to script the use of every individual spell in Amnaf's list..
+    end
+
```

Comment:
To my knowledge that doesn't work with onSpellCast like it does with onMobWeaponSkill. We had onMonsterMagicPrepare (which lacks a spellID/spell argument) and onSpellPrecast (which lacks a target argument). I think the best solution will be to add anotehr form of resetEmnity that doesn't require target input so I can just say `mob:resetAllEnmity();` in onSpellPrecast in the mobs script. But that doesn't exist yet.